From 23d8fbf825273bccc309ec5b569552ffa75e5d52 Mon Sep 17 00:00:00 2001
From: Florian Sylvestre <fsylvestre@baylibre.com>
Date: Tue, 25 Apr 2023 15:03:36 +0200
Subject: [PATCH 02/13] machines: cortex-a: Add helpers to access cp15
 registers.

---
 machines/cortex-a/emblib.h | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/machines/cortex-a/emblib.h b/machines/cortex-a/emblib.h
index e949ef96..31b7934d 100644
--- a/machines/cortex-a/emblib.h
+++ b/machines/cortex-a/emblib.h
@@ -15,3 +15,31 @@
 
 #define readFromAddress(address) \
   *ADDR(address)
+
+#define cp15_read32(op1, CRn, CRm, op2, val)                \
+{                                                           \
+    __asm__ volatile (                                      \
+        "mrc p15, " #op1 ", %0, c" #CRn ", c"               \
+        #CRm ", " #op2 : "=r" (val) :: "memory");           \
+}
+
+#define cp15_write32(op1, CRn, CRm, op2, val)               \
+{                                                           \
+    __asm__ volatile (                                      \
+        "mcr p15, " #op1 ", %0, c" #CRn ", c"               \
+        #CRm ", " #op2 :: "r" (val) : "memory");            \
+}
+
+#define cp15_read64(op1, CRm, val)                          \
+{                                                           \
+    __asm__ volatile (                                      \
+        "mrrc p15, " #op1 ", %Q0, %R0, c"                   \
+        #CRm : "=r" (val) :: "memory");                     \
+}
+
+#define cp15_write64(op1, CRm, val)                         \
+{                                                           \
+    __asm__ volatile (                                      \
+        "mcrr p15, " #op1 ", %Q0, %R0, c"                   \
+        #CRm :: "r" (val) : "memory");                      \
+}
-- 
2.34.1

