From ebb46850732ea2430ff189a0a412f5b7f1a59837 Mon Sep 17 00:00:00 2001
From: Valerio Setti <vsetti@baylibre.com>
Date: Mon, 3 Jul 2023 17:56:19 +0200
Subject: [PATCH 45/70] phy: fix MII access

Signed-off-by: Valerio Setti <vsetti@baylibre.com>
---
 .../cortex-a/armv8/spider/ethernet/rswitch.c   | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/examples/cortex-a/armv8/spider/ethernet/rswitch.c b/examples/cortex-a/armv8/spider/ethernet/rswitch.c
index c653266d..64e944aa 100644
--- a/examples/cortex-a/armv8/spider/ethernet/rswitch.c
+++ b/examples/cortex-a/armv8/spider/ethernet/rswitch.c
@@ -779,31 +779,31 @@ static int rswitch_phy_init(struct rswitch_etha *etha)
     int ret;
 
     /* Reset */
-    CHECK_RET(rswitch_mii_read(etha, etha->port_num, 1, 0xC04A, &reg_data));
+    CHECK_RET(rswitch_mii_read(etha, 1, etha->port_num + 1, 0xC04A, &reg_data));
     reg_data |= BIT(15);
-    CHECK_RET(rswitch_mii_write(etha, etha->port_num, 1, 0xC04A, &reg_data));
-    CHECK_RET(rswitch_mii_write(etha, etha->port_num, 1, 0xC04A, &reg_data)); /* MCAL does it twice... */
+    CHECK_RET(rswitch_mii_write(etha, 1, etha->port_num + 1, 0xC04A, &reg_data));
+    CHECK_RET(rswitch_mii_write(etha, 1, etha->port_num + 1, 0xC04A, &reg_data)); /* MCAL does it twice... */
 
     /* Check mode */
-    CHECK_RET(rswitch_mii_read(etha, etha->port_num, 1, 0xC04A, &reg_data));
+    CHECK_RET(rswitch_mii_read(etha, 1, etha->port_num + 1, 0xC04A, &reg_data));
     if ((reg_data & 0x7) != 0x4 ) { /* 4 stands for SGMII */
         reg_data &= ~0x7;
         reg_data |= BIT(15) | 0x4;
-        CHECK_RET(rswitch_mii_write(etha, etha->port_num, 1, 0xC04A, &reg_data));
+        CHECK_RET(rswitch_mii_write(etha, 1, etha->port_num + 1, 0xC04A, &reg_data));
 
         /* Run SERDES Init */
-        CHECK_RET(rswitch_mii_read(etha, etha->port_num, 1, 0x800F, &reg_data));
+        CHECK_RET(rswitch_mii_read(etha, 1, etha->port_num + 1, 0x800F, &reg_data));
         reg_data |= BIT(15) | BIT(13);
-        CHECK_RET(rswitch_mii_write(etha, etha->port_num, 1, 0x800F, &reg_data));
+        CHECK_RET(rswitch_mii_write(etha, 1, etha->port_num + 1, 0x800F, &reg_data));
         
         reg_data = 0x0U;
         do {
-            CHECK_RET(rswitch_mii_read(etha, etha->port_num, 1, 0x800F, &reg_data));
+            CHECK_RET(rswitch_mii_read(etha, 1, etha->port_num + 1, 0x800F, &reg_data));
         } while (reg_data & BIT(15));
 
         /* Auto SERDES Init Disable */
         reg_data &= ~BIT(13);
-        CHECK_RET(rswitch_mii_write(etha, etha->port_num, 1, 0x800F, &reg_data));
+        CHECK_RET(rswitch_mii_write(etha, 1, etha->port_num + 1, 0x800F, &reg_data));
     }
 
     return 0;
-- 
2.34.1

