From 1ea74de1cb62b38ceb2a79dc1342a9d31c6b2b18 Mon Sep 17 00:00:00 2001
From: Florian Sylvestre <fsylvestre@baylibre.com>
Date: Thu, 21 Sep 2023 17:27:51 +0200
Subject: [PATCH 61/70] core: fix 'tpl_task_events_table' section description

---
 os/tpl_os_task_kernel.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/os/tpl_os_task_kernel.h b/os/tpl_os_task_kernel.h
index 11f80039..5166860e 100644
--- a/os/tpl_os_task_kernel.h
+++ b/os/tpl_os_task_kernel.h
@@ -49,7 +49,7 @@ struct TPL_TASK_EVENTS {
 typedef struct TPL_TASK_EVENTS tpl_task_events;
 
 
-#define OS_START_SEC_VAR_UNSPECIFIED
+#define OS_START_SEC_CONST_UNSPECIFIED
 #include "tpl_memmap.h"
 
 #if EXTENDED_TASK_COUNT > 0
@@ -60,7 +60,7 @@ extern CONSTP2VAR(tpl_task_events, AUTOMATIC, OS_APPL_DATA)
         tpl_task_events_table[EXTENDED_TASK_COUNT];
 #endif /* EXTENDED_TASK_COUNT */
 
-#define OS_STOP_SEC_VAR_UNSPECIFIED
+#define OS_STOP_SEC_CONST_UNSPECIFIED
 #include "tpl_memmap.h"
 
 #define OS_START_SEC_CODE
-- 
2.34.1

