From b441d938de16180a14b64e8fbb4a9be9be7f50df Mon Sep 17 00:00:00 2001
From: Duy Dang <duy.dang.yw@renesas.com>
Date: Fri, 23 Jun 2023 15:09:25 +0900
Subject: [PATCH 06/23] whitebox: Add S4SK to build script

Signed-off-by: Duy Dang <duy.dang.yw@renesas.com>
---
 build_whitebox_v3.0.sh | 27 ++++++++++++++++++---------
 1 file changed, 18 insertions(+), 9 deletions(-)

diff --git a/build_whitebox_v3.0.sh b/build_whitebox_v3.0.sh
index 389b4c5..ea4ab81 100755
--- a/build_whitebox_v3.0.sh
+++ b/build_whitebox_v3.0.sh
@@ -14,7 +14,10 @@ BUILD_PATTERS_NUM=$((${#BUILD_PATTERNS[@]} / 3  -1))
 
 Usage() {
     echo "Usage:"
-    echo "    $0 [option]"
+    echo "    $0 [board] [option]"
+    echo "board:"
+    echo "    - spider: for S4 Spider"
+    echo "    - s4sk: for S4 Starter Kit"
     echo "option:"
     for key in `seq 1 ${BUILD_PATTERS_NUM}`; do
         echo -ne "    -${key}: "
@@ -25,11 +28,17 @@ Usage() {
     echo "    -h: Show this usage"
 }
 
+# Board selection is required
+if [[ $# < 1 ]] || [[ "$1" != "spider" ]] && [[ "$1" != "s4sk" ]]; then
+    echo "Please select a board to build"
+    Usage; exit
+fi
+
 # Without option, pattern 1 is selected by default.
-if [[ $# < 1 ]]; then
-    $0 -1
+if [[ $# < 2 ]]; then
+    $0 $1 -1
     exit
-elif [[ $# > 1 ]]; then # option can be used only one
+elif [[ $# > 2 ]]; then # option can be used only one
     Usage; exit
 fi
 
@@ -57,13 +66,13 @@ esac
 cd ${BASE_DIR}/realtime_cpu
 case $CR52 in
     "Trampoline") ./build_trampoline.sh -c;;
-    "Zephyr")     ./build_zephyr.sh -c;;
+    "Zephyr")     ./build_zephyr.sh $1 -c;;
 esac
 
 # Build CA55
 cd ${BASE_DIR}/application_cpu
 case $CA55 in
-    "Linux") ./build_ca55.sh -c;;
+    "Linux") ./build_ca55.sh $1 -c;;
 esac
 
 # deploy images
@@ -72,8 +81,8 @@ mkdir -p deploy
 cp mcu/deploy/g4mh.srec deploy/App_CDD_ICCOM_S4_Sample_G4MH.srec
 cp realtime_cpu/deploy/cr52.srec deploy/App_CDD_ICCOM_S4_Sample_CR52.srec
 cp application_cpu/work/full.img.gz deploy/
-cp application_cpu/work/yocto/build-domd/tmp/deploy/images/spider/bl31-spider.srec deploy/bl31.srec
-cp application_cpu/work/yocto/build-domd/tmp/deploy/images/spider/tee-spider.srec deploy/tee.srec
-cp application_cpu/work/yocto/build-domd/tmp/deploy/images/spider/u-boot-elf.srec deploy/u-boot-elf.srec
+cp application_cpu/work/yocto/build-domd/tmp/deploy/images/${1}/bl31-${1}.srec deploy/bl31.srec
+cp application_cpu/work/yocto/build-domd/tmp/deploy/images/${1}/tee-${1}.srec deploy/tee.srec
+cp application_cpu/work/yocto/build-domd/tmp/deploy/images/${1}/u-boot-elf.srec deploy/u-boot-elf.srec
 ls -l deploy
 
-- 
2.34.1

