From c0fe2749f66d94f4330e6c505d24146aa3b4d4f5 Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Fri, 31 Mar 2023 13:44:46 +0900
Subject: [PATCH 26/27] examples: iccom: Add optimaization flag

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 examples/renesas/iccom/iccom.oil | 1 +
 1 file changed, 1 insertion(+)

diff --git a/examples/renesas/iccom/iccom.oil b/examples/renesas/iccom/iccom.oil
index c20e1ea..b31862a 100644
--- a/examples/renesas/iccom/iccom.oil
+++ b/examples/renesas/iccom/iccom.oil
@@ -24,6 +24,7 @@ CPU iccom {
       LDFLAGS="-debug -nocompress -NOOPtimize -memory=high -nologo -SHow=ALL";
       CFLAGS="-Xcpu=g4mh -g -g_line -Xfxu=off -Xasm_path=.";
       CFLAGS="-Xfloat=soft -Xdbl_size=8 -Xround=nearest";
+      CFLAGS="-Ospeed -Oinline_size";
       LINKER = "rlink";
       SYSTEM = CMAKE;
     };
-- 
2.34.1

