From f623acb08c80e7e407d4cb5bba381e2e397f427e Mon Sep 17 00:00:00 2001
From: Tsutomu Muroya <tsutomu.muroya.jy@bp.renesas.com>
Date: Thu, 27 Apr 2023 11:24:20 +0900
Subject: [PATCH 29/29] Change of G4MH start address

Signed-off-by: Tsutomu Muroya <tsutomu.muroya.jy@bp.renesas.com>
---
 machines/renesas/g4mh/rcar_s4.clnk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/machines/renesas/g4mh/rcar_s4.clnk b/machines/renesas/g4mh/rcar_s4.clnk
index 4e42df86..c4488317 100755
--- a/machines/renesas/g4mh/rcar_s4.clnk
+++ b/machines/renesas/g4mh/rcar_s4.clnk
@@ -1,5 +1,5 @@
 -library=lib\v850e3v5\rhs8n.lib
 -library=lib\v850e3v5\libmalloc.lib
--start=RESET_PE0,EIINTTBL_PE0/200,.const,.INIT_DSEC.const,.INIT_BSEC.const,.text.cmn,syscall.text,.text,.data/00010000,.data.R,.bss,.stack.bss/FDE00000
+-start=RESET_PE0,EIINTTBL_PE0/400,.const,.INIT_DSEC.const,.INIT_BSEC.const,.text.cmn,syscall.text,.text,.data/00010000,.data.R,.bss,.stack.bss/FDE00000
 -fsymbol=.text.cmn
 -rom=.data=.data.R
-- 
2.25.1

